src/odemis/gui/comp/miccanvas.py

```diff
@@ -1268,12 +1268,18 @@ def setView(self, view, tab_data):
         self.view = view
         self._tab_data_model = tab_data
 
-    def set_2d_data(self, im_data, unit_x=None, unit_y=None, range_x=None, range_y=None):
+    def set_2d_data(self, im_data, unit_x=None, unit_y=None, range_x=None, range_y=None, xflip=False, yflip=False):
```

Comment:
Just accept a single parameter: `flip`, which can contain `wx.HORIZONTAL | wx.VERTICAL`. Note that these two values can be simultaneously set.
Then, you pass it as-is to `set_image`. Up to the caller to compute the right values.